src/reagent/impl/util.cljs

```diff
@@ -24,18 +24,29 @@
 (def dont-camel-case #{"aria" "data"})
 
 (defn capitalize [s]
-  (if (< (count s) 2)
+  (if (< (.-length s) 2)
     (string/upper-case s)
     (str (string/upper-case (subs s 0 1)) (subs s 1))))
 
 (defn dash-to-prop-name [dashed]
   (if (string? dashed)
     dashed
     (let [name-str (name dashed)
-          [start & parts] (string/split name-str #"-")]
-      (if (dont-camel-case start)
-        name-str
-        (apply str start (map capitalize parts))))))
+          [start & parts] (.split name-str #"-")]
+      (cond
+        (contains? dont-camel-case start) name-str
+
+        (not (nil? parts))
```

Comment:
I think this could be just `parts`?

Reply:
Leaving just `parts` is going to generate truthy check around it, simpler form would be `(some? parts)`. I'd prefer to explicitly coerce `parts` to `boolean` instead of type hinting it with `^boolean`, so checked if can be eliminated.